### ISSUE TITLE
Compiler crash on calling a variable

### ISSUE THREAD

**Issue by Bara** [alliedmodders/sourcepawn#750]
Test case:
```sp
public void main()
{
    int test;
    test();
}

```

Is crashes on 1.11 - 6837

seems 1.10 - 6522 can handle this:
> test.sp(4) : error 012: invalid function call, not a valid address
test.sp(4) : error 029: invalid expression, assumed zero
test.sp(4) : error 215: expression has no effect
test.sp(4) : fatal error 190: too many error messages on one line

**Comment by Fyren** [comment]
This dies because the `CallExpr` has a null `sym`. https://github.com/alliedmodders/sourcepawn/blob/84adc2f3c518851496fd72aca9909510b9bc1370/compiler/code-generator.cpp#L1813-L1815 

`CheckCallExpr` for this `CallExpr` does return false, but it's bubbled up to `CheckBlockStmt` which doesn't check it.  https://github.com/alliedmodders/sourcepawn/commit/6ccea09a8dbb40cb89617e1f06fd3a956d76b767 is one way of fixing it, but maybe it should error even earlier?

**Comment by dvander** [comment]
Yeah, BindCallTarget is returning nullptr with no report. Whoops.

**Comment by dvander** [comment]
Thanks, this has been fixed on master.

**Comment by Jackzmc** [comment]
Was this fixed for 1.11? I'm having a compiler crash caused by calling a methodmap property as a function.

No crash:  `int prev = Game.MapTime;`
Crash: `int prev = Game.MapTime();`

Compiler: **SourcePawn Compiler 1.11.0.6905**
Seg faults with terminated with exit code: 3221225477.

**Comment by dvander** [comment]
It looks like there is still an issue here.
